### PR TITLE
Building ratings component

### PR DESCRIPTION
**Pull request by harrisonywu**
@krise3 Let me know how this looks.

### CODE REVIEW

File: client/src/components/App.jsx [hunk 1]
```diff
@@ -11,10 +11,12 @@ class App extends React.Component {
...
       currentSearchTerm: null,
+      allAverageRatings: {},
     }
     this.searchSubmit = this.searchSubmit.bind(this);
     this.searchBarTextChange = this.searchBarTextChange.bind(this);
     this.calculateAverageRating = this.calculateAverageRating.bind(this);
+    this.storeAllAverageRatings = this.storeAllAverageRatings.bind(this);
   }
   
   componentDidMount() {
```
Comment (krise3) [inline client/src/components/App.jsx]: Would be a bit nicer if this function was written without storing context in a constant

File: client/src/components/App.jsx [hunk 2]
```diff
@@ -59,17 +62,36 @@ class App extends React.Component {
       combinedRatingTotal += this.state.allReviews[i][ratingCategory];
     }
     const averageRating = combinedRatingTotal / numberOfReviews;
-    const averageRatingToClosestHalfRating = Math.ceil(averageRating*2)/2
-    return averageRatingToClosestHalfRating;
+    const nearestHalfRating = Math.ceil(averageRating*2)/2
```
Comment (krise3) [inline client/src/components/App.jsx]: Nice variable name refactor!

File: client/src/components/Ratings.jsx [hunk 3]
```diff
@@ -4,15 +4,48 @@ class Ratings extends React.Component {
...
+      halfStar += '★'
+    };
+
+    // stars now have their own individual classes, find out how to append the half star to the end of the full star.
+    return (
+      <div>
+        <div className="full-star">
+          {fullStars}
+        </div>
+        <div className="half-star">
```
Comment (krise3) [inline client/src/components/Ratings.jsx]: Interesting. Just out of curiosity, what was the basis for creating the half-star class?

Comment (harrisonywu) [reply client/src/components/Ratings.jsx]: @krise3 I'm going to use those to modify how the CSS looks. full-star will look different than half-star.

**Pull request merged**